src/main/java/no/finn/unleash/repository/FeatureToggleRepository.java

```diff
@@ -40,7 +39,14 @@ public FeatureToggleRepository(
         this.toggleFetcher = toggleFetcher;
         this.eventDispatcher = new EventDispatcher(unleashConfig);
 
-        toggleCollection = toggleBackupHandler.read();
+        this.toggleCollection = toggleBackupHandler.read();
+        if (toggleCollection != null && toggleCollection.getFeatures().isEmpty()) {
+            if (unleashConfig.getToggleBootstrapProvider() != null) {
+                toggleCollection =
+                        new ToggleBootstrapHandler(unleashConfig)
```

Comment:
I feel it would be more readable if we already have constructed the `ToggleBootstrapHandler` even though it wont be needed if a `ToggleBootstrapProvider` is not provided.

Reply:
I was trying to avoid unnecessary initialization, but I'd be fine with just instantiating the BootstrapHandler and let it handle whether or not its provider is initialized. I'll change the code.